zowe/zowe-cli
# Add --ignore-not-found flag logic to zosfilesBase.handler and update command definitions and en.ts for delete operations

**What it Does**
This PR implements the `--ignore-not-found` flag for delete operations across DS, USS, zFS, and VSAM by:
- Adding logic in the `ZosFilesBaseHandler` to handle `--ignore-not-found` and suppress errors when a file or data set does not exist (HTTP 404 or IDC3012I).
  - The IDC3012I code is an IBM z/OS error message that indicates a VSAM entry does not exist in the catalog
- Updating the command definitions to include the `--ignore-not-found` flag.
- Modifying `en.ts` to reflect the new flag in descriptions and examples.
 
![image](https://github.com/user-attachments/assets/56191d2c-c7a3-4292-9415-d7f2dfa6b33f)

**How to Test**
Verify that the `--ignore-not-found` flag works for all delete operations (DS, USS, zFS, VSAM) by attempting to delete things that don't exist. Delete non-existent files/data sets with and without `--ignore-not-found`.

1. Test commands with --for-sure and --ignore-not-found flag (-fi)

| Command                                                | Type  |
|--------------------------------------------------------|-------|
| `zowe zos-files delete data-set "ibmuser.cntl" -fi`     | DS    |
| `zowe zos-files delete uss-file "/a/ibmuser/testcases" -fi` | USS   |
| `zowe zos-files delete zos-file-system "HLQ.MYNEW.ZFS" -fi` | zFS   |
| `zowe zos-files delete data-set-vsam "ibmuser.AAA.**.FFF" -fi` | VSAM  |


2. Test commands with just the --for-sure flag (-f)

| Command                                                | Type  | Expected Error Message                                          |
|--------------------------------------------------------|-------|-----------------------------------------------------------------|
| `zowe zos-files delete data-set "ibmuser.cntl" -f`      | DS    | "Data set not cataloged"                 |
| `zowe zos-files delete uss-file "/a/ibmuser/testcases" -f` | USS   | "No such file or directory"             |
| `zowe zos-files delete zos-file-system "HLQ.MYNEW.ZFS" -f` | zFS   | "NOT FOUND"         |
| `zowe zos-files delete data-set-vsam "ibmuser.AAA.**.FFF" -f` | VSAM  | "NOT DELETED"       |



**Review Checklist**
I certify that I have:
- [x] I added/updated automated tests.
- [x] I followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md).


**Additional Comments**
This update is for all delete operations except MDS (which does not currently support this flag)
